ovs/extensions/healthcheck/alba/alba_health_check.py

```diff
@@ -370,81 +366,78 @@ def check_backend_asds(logger, disks, backend_name, config):
         logger.info("Checking seperate ASD's for backend '{0}':".format(backend_name), 'check_asds')
 
         # check if disks are working
-        if len(disks) != 0:
-            # Map long id to ip
-            osd_mapping = {}
+        if len(disks) == 0:
+            return workingdisks, defectivedisks
+        # Map long id to ip
+        osd_mapping = {}
+        try:
+            for osd in AlbaCLI.run(command='list-osds', config=config):
+                osd_mapping[osd.get('long_id')] = osd.get('ips')[0]
+        except RuntimeError as ex:
+            logger.failure("Could not fetch osd list from Alba. Got {0}".format(str(ex)))
+            return None
+        for disk in disks:
+            key = 'ovs-healthcheck-{0}'.format(str(uuid.uuid4()))
```

Comment:
Its more readable to retrieve all the disk stuff you need here once, instead off in the entire code to use ```disk.get(<key>)
So like this:
```
for disk in disks:
    disk_status = disk.get('status')
    disk_asd_id = disk.get('asd_id')
    disk_status_detail = disk.get('status_detail')
    ...
```

Also you shouldn't do ```.get``` on each key. In current implementation these keys WILL be present and the healthcheck should fail if these keys would disappear or be renamed in future versions, so you can adjust the healthcheck accordingly. Now when something would change, the healthcheck would still work, but give incorrect output